app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/address/MixedUsePropertyDetailsController.scala

```diff
@@ -0,0 +1,408 @@
+
+        returnsService
+          .storeDraftReturn(
+            updatedDraftReturn,
+            journey.journey.subscribedDetails.cgtReference,
+            journey.journey.agentReferenceNumber
+          )
+          .map(_ => journey.journey.copy(draftReturn = updatedDraftReturn))
+
+    }
+
+  def isATrust(journey: EnteringSingleMixedUsePropertyDetails): Boolean =
+    journey.journey.subscribedDetails.isATrust
+
+  protected lazy val isUkCall: Call                                              = enterPostcodeCall
+  protected lazy val isUkSubmitCall: Call                                        = enterPostcodeCall
+  protected lazy val enterNonUkAddressCall: Call                                 = enterPostcodeCall
+  protected lazy val enterNonUkAddressSubmitCall: Call                           = enterPostcodeCall
+  protected lazy val backLinkCall: EnteringSingleMixedUsePropertyDetails => Call = _ => enterPostcodeCall
+  protected lazy val ukAddressNotAllowedExitPageCall: Option[Call]               = None
+
+  protected lazy val enterUkAddressCall: Call       = addressRoutes.MixedUsePropertyDetailsController.enterUkAddress()
+  protected lazy val enterUkAddressSubmitCall: Call =
+    addressRoutes.MixedUsePropertyDetailsController.enterUkAddressSubmit()
+  protected lazy val enterPostcodeCall: Call        = addressRoutes.MixedUsePropertyDetailsController.enterPostcode()
+  protected lazy val enterPostcodeSubmitCall: Call  =
+    addressRoutes.MixedUsePropertyDetailsController.enterPostcodeSubmit()
+  protected lazy val selectAddressCall: Call        = addressRoutes.MixedUsePropertyDetailsController.enterUkAddress()
+  protected lazy val selectAddressSubmitCall: Call  = addressRoutes.MixedUsePropertyDetailsController.selectAddress()
+  protected lazy val continueCall: Call             = addressRoutes.MixedUsePropertyDetailsController.checkYourAnswers()
+
+  def singleMixedUseGuidance(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val backLink =
+          r.answers
+            .fold(
+              _ => controllers.returns.routes.TaskListController.taskList(),
+              _ => routes.MixedUsePropertyDetailsController.checkYourAnswers()
+            )
+        Ok(
+          singleMixedUseGuidance(
+            backLink
+          )
+        )
+      }
+    }
+
+  def singleMixedUseGuidanceSubmit(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, _) =>
+        Redirect(enterPostcodeCall)
+      }
+    }
+
+  def enterDisposalValue(): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withValidJourney(request) { (_, r) =>
+        val answers       = r.answers
+        val backLink      = routes.MixedUsePropertyDetailsController.singleMixedUseGuidance()
```

Comment:
this should be the cya endpoint if the answers are complete

Reply:
and should point to the same back link location defined  in the submit endpoint when it's incomplete